### ISSUE TITLE
Keyboard indicator 

### ISSUE THREAD

**Issue by bil-elmoussaoui** [numixproject/numix-icon-theme#271]
Hello, i've a little problem,
i'm using Numix Icon Theme on my Elementary OS 0.3 Beta 1 
![capture du 2015-02-01 22 16 51](https://cloud.githubusercontent.com/assets/7660997/5993904/228ccd1a-aa60-11e4-8fcb-eae300174ca7.png)
this is my keyboard indicator for the french language ; when i add an other language like english :  and i chose this one the icon became like this ? but only for the english 
![capture du 2015-02-01 22 17 07](https://cloud.githubusercontent.com/assets/7660997/5993905/3d22a97e-aa60-11e4-90a3-e44b3ac7ac1d.png)
i added one more language to test it (arabic) and i dont have this problem? 
![capture du 2015-02-01 22 17 30](https://cloud.githubusercontent.com/assets/7660997/5993907/587824ec-aa60-11e4-9a26-9cf9186d466b.png)

I dont know if i have to give you more informations? 
Thanks

**Comment by dirtydancing** [comment]
Currently, Numix base comprises only a couple of indicator-keyboard icons, i.a. for EN. As elementary OS Freya (Beta) is using the canvas size 24x24 in the wingpanel, for EN the current Numix base icon is used. For the other languages in your screenshots above, as these icons are white and smaller, to me this looks like they are taken from elementary icon theme (which seems to be inherited in your settings).

I am not sure why the "EN" looks so greyed out in Numix base in size 24, this is different e.g. in size 22. In the following screenshots you can see which indicator-keyboard icons are currently in Numix base, and that the colours differ inbetween size 24x24 (e.g. elementary OS Freya) and 22x22 (e.g. Ubuntu).

`Numix/24x24/status` (EN marked, somewhat "greyed out", like in your screenshot):

![numix_24x24_status](https://cloud.githubusercontent.com/assets/7164227/6433349/b3b91862-c06c-11e4-8df4-79538ca5b6e4.png)

`Numix/22x22/status` (EN marked):

![numix_22x22_status](https://cloud.githubusercontent.com/assets/7164227/6433354/d6ba87d8-c06c-11e4-880b-87d0291857a3.png)

**Comment by dirtydancing** [comment]
In addition, the indicator-keyboard icons in Numix base are also "greyed out" in size 16x16, just as in 24x24.

`Numix/16x16/status` (EN marked):

![16x16_status](https://cloud.githubusercontent.com/assets/7164227/6482231/32e7759a-c264-11e4-8d4d-3adac92f8769.png)

This is visible e.g. in Ubuntu 14.10: in the panel, the Numix base 22x22 En icon is used and displayed correctly, whereas in the dropdown menu the "greyed out" Numix base 16x16 En icon used:

![ubuntu1410_en](https://cloud.githubusercontent.com/assets/7164227/6482258/641a8ef4-c264-11e4-9e61-6cfc6c5d25b9.png)

I have noticed that there is a pull request for size 16x16 (I have not tested that one), https://github.com/numixproject/numix-icon-theme/pull/272

It would probably be a good idea to keep in mind the consistency between sizes 16x16, 22x22 and 24x24.

As far as I can tell, as these are not "-symbolic" named icons, they are not influenced by the respective GTK theme and they are always displayed in the colour that they are designed in.

This would mean that for Numix base, all the indicator-keyboard icons would have to be designed in the colour that Numix base uses for the panel status icons.

**Comment by bil-elmoussaoui** [comment]
the english icon is a little bigger than the others and fit better with Numix base panel status icons
![screenshot from 2015-03-04 14 10 58](https://cloud.githubusercontent.com/assets/7660997/6484032/560e601a-c278-11e4-80ae-e9321a162fc5.png)
another screenshot with a different language 
![screenshot from 2015-03-04 14 11 17](https://cloud.githubusercontent.com/assets/7660997/6484042/67021402-c278-11e4-96bf-b1126bf37b13.png)

**Comment by dirtydancing** [comment]
@bil-elmoussaoui Is this Ubuntu in your screenshot above? (and not elementary OS). I am asking because the `En` indicator icon is displayed correctly from current Numix base, which means that this should be a current 22x22 status icon.

For `Fr` there is no Numix base icon yet. I have been wondering why this icon is displayed at all, because Numix icon theme inherits gnome and hicolor icon themes, and I think neither of these contain indicator-keyboard icons for Ubuntu. In your screenshot, the `Fr` icon appears to be neither from Numix base nor e.g. from Ubuntu-mono-dark icon theme nor from elementary icon theme (?).

**Comment by bil-elmoussaoui** [comment]
The screenshot is from Ubuntu 14.04 (installed it to test Numix base theme on it)

**Comment by dirtydancing** [comment]
As to the exact colour of the panel icons also cf. https://github.com/numixproject/numix-icon-theme/issues/282

**Comment by dirtydancing** [comment]
Maybe it would be possible to have an automated way to generate all the Numix indicator-keyboard icons for all languages? I am not sure though if that could create the kind of drop shadow that Numix is usually using for the fixed size panel icons. Just an idea.

**Comment by wa4557** [comment]
I created a pull-request with all the languages c.f. #283

**Comment by dirtydancing** [comment]
@wa4557 Wow, phantastic! Did you auto-generate all these or did you create them individually?

Anyways, in your pull request https://github.com/numixproject/numix-icon-theme/pull/283 you mentioned a couple of aspects that fit into this issue thread, so I am addressing them here.

**1) Colour:** You write "I used #dcdcdc instead of #dbdbdb as base color since it is the one used most of the times"

Am I correct in assuming that instead of #dbdbdb you were referring to #dedede? Cf. also https://github.com/numixproject/numix-icon-theme/issues/282

The call as to what colour exactly the grey colour for the panel should be is not up to me. All of these are great, but I figure that the grey colour should be consistent. There is already a "Numix white" at value #f9f9f9. So maybe this would be a good opportunity to introduce a "Numix grey"? :-) @Foggalong?

**2) Sizes:** You write: "I deleted the keyboard indicators from 16x16 and 24x24 since they were wrong anyway and are useless imo. Prove me wrong and I'll add proper sized icons there as well"

Let me try and prove you wrong then :-). The sizes 16x16 and 24x24 are also needed. The size 22x22 is the correct one for the Ubuntu panel (canvas size 22x22). At the same time, once this icon exists in 22x22, usually this will also be taken for other canvas sizes and scaled - unless there is also this icon in 16x16 and 24x24.

Size **24x24** is used e.g. in the panel in elementary OS Freya (currently Beta 2). If there is no icon in 24x24, the icon from 22x22 will get scaled and will be too big and possibly blurred. Cf. for this scenario also this issue report concerning several app status icons in Xubuntu: https://github.com/numixproject/numix-icon-theme-circle/issues/1601

To illustrate, I have installed your "Numix proposed" version of Numix base in Ubuntu 14.10 (real hardware, above) as well as in elementary OS Freya (VM, below). The first screenshot shows the current status for De (seems to be somehow auto-generated in dark grey if no icon is found). The icon sizes are equal. The second screenshot shows the same, this time including your pull request. The indicator-keyboard icon is correctly displayed in Ubuntu (canvas size 22x22), but too big and somewhat distorted in elementary OS Freya (canvas size 24x24, scaling necessary).

Numix
![elementary_numix](https://cloud.githubusercontent.com/assets/7164227/6542305/ccb48fcc-c4f4-11e4-9adb-33ffec0119d9.png)

Numix proposed
![elementary_numix-proposed](https://cloud.githubusercontent.com/assets/7164227/6542306/df52a7c2-c4f4-11e4-9ff0-1f32b1620db7.png)

The icons for size 24x24 would be in the same size as the icons for 22x22, only the canvas would be 24x24 instead of 22x22. That way, they are displayed in the same size in both cases, just as all the other system status icons in the two screenshots above.

Now on to size **16x16**: This size is used e.g. in Ubuntu for the dropdown menu. Again, if there is only an icon in 22x22, that one will be taken, scaled down and possibly blurred. The first screenshot below shows Ubuntu 14.10 with default Ubuntu-mono-dark icon theme. In the dropdown menu, De and En are displayed fine. The second screenshot shows Ubuntu 14.10 with "Numix proposed", including your pull request. The icons in the dropdown menu there are somewhat distorted again, visible esp. for De. Depending on the language letters (and possibly additional numbers), this might get worse. It merely seems to be a coincidence that En is looking just fine in this case.

Ubuntu-mono-dark
![ubuntu_ubuntu-mono-dark](https://cloud.githubusercontent.com/assets/7164227/6542337/c7171ad4-c4f5-11e4-99c9-abfd184d6c06.png)

Numix proposed
![ubuntu_numix-proposed](https://cloud.githubusercontent.com/assets/7164227/6542338/ce01ae68-c4f5-11e4-8c92-c3ebc2752596.png)

Convinced? :-)

**Comment by wa4557** [comment]
Yeah, I've written a small python-script to generate them automatically...

I've used #dcdcd instead of #dedede because almost all other panel icons are coloured this way, frankly there's not much difference between those two:)

I see, so if I'm correct: 24x24 with same font-size but bigger canvas, and 16x16 scaled downaccordingly?
The 24x24 ones are easy to do, for the 16x16 ones I have to think about sth clever but is definitely doable, I'll give it a try

**Comment by dirtydancing** [comment]
Yeah, I agree that these two greys look basically the same, at least if you do not know that there are two diferent colours in the panel :-), but I keep on coming across Numix panel icons in #dedede, and consistency is the thing here.

And well, I just checked, and Ubuntu-mono-dark seems to have indicator-keyboard icons only in size 22x22 (?) :-) Ubuntu would not need size 24x24, for sure. And elementary icon theme encompasses sizes 16x16 and 24x24.

Whatever is going on there exactly, the point of having sizes 16x16 and 24x24 in addition to size 22x22 is still valid: Numix icon theme is not only catering specifically to Ubuntu.

To your question: Yes, 24x24 with same font-size and all the same, just a bigger canvas. As for 16x16, I am not sure if there is a guideline for the scaling down. I would guess indeed just simply scaled down from 22x22, but for size 16x16 icons I have come across different approaches; sometimes they are larger relative to the canvas, but not necessarily for the panel icons. @Foggalong?

**Comment by wa4557** [comment]
Ok I've added the 24x24 ones; I'm not sure how the 16x16 should look like, so I won't do them for now -> I'm not a designer, just a programmer :)

**Comment by wa4557** [comment]
Nevermind I've added the 16x16 icons as well, if you don't like them tell me and I'll remove them. @dirtydancing: can you try out? I'm right now working from a mac -> not themeable

**Comment by dirtydancing** [comment]
Even more phantastic now! Here come some screenshots to illustrate:

First of all, testing in Ubuntu 14.10 if really 22x22 and 16x16 are used in the presence of 16, 22 and 24 (yes):
![ubuntu_16_22](https://cloud.githubusercontent.com/assets/7164227/6543389/1cd5aebc-c51a-11e4-92be-9d9c0fde2328.png)

Comparison in Ubuntu 14.10 between proposed and proposed_NEW. In the dropdown menu, now the 16x16 icon is showing, the text is better placed and you can see the rounded edges/drop shadow.

The way the "De" looks a bit close together I attribute to the limited resolution of my screen (1280x800). The icon itself is fine.

proposed (as already above)
![ubuntu_proposed](https://cloud.githubusercontent.com/assets/7164227/6543399/7fbd929c-c51a-11e4-9c0e-ef84f7936dff.png)

proposed_NEW
![ubuntu_proposed_new](https://cloud.githubusercontent.com/assets/7164227/6543400/87559ec8-c51a-11e4-904a-ee2f720a2f34.png)

As to elementary OS Freya, with the added 24x24 icons this is now solid as well:

proposed (as already above)
![elementary_proposed](https://cloud.githubusercontent.com/assets/7164227/6543402/a07ef9f8-c51a-11e4-9c9f-a218bca157b8.png)

proposed_NEW
![elementary_proposed_new](https://cloud.githubusercontent.com/assets/7164227/6543404/a93a52d6-c51a-11e4-9da4-a86123f41857.png)

And just to show from where this has come today in elementary OS Freya with dropdown menu (all of the screenshots here with Numix GTK theme):

Numix current
![elementary_dropdown_current](https://cloud.githubusercontent.com/assets/7164227/6543412/d3959c16-c51a-11e4-9952-db17c071573b.png)

Numix proposed
![elementary_dropdown_proposed](https://cloud.githubusercontent.com/assets/7164227/6543413/d98acf1a-c51a-11e4-98e9-6673eaf64d4f.png)

Numix proposed_NEW
![elementary_dropdown_proposed_new](https://cloud.githubusercontent.com/assets/7164227/6543414/de9a6768-c51a-11e4-8b03-8ba7cf2bafa3.png)

In summary: this is spot on! I saw that you even included a corresponding Numix-Light variant, and when I had a quick glance at that in combination with Radiance GTK theme in Ubuntu, all looked just fine as well.

All of this is fixing a very visible issue in Numix for those who are making use of the keyboard language indicator in Ubuntu & Co.

**Comment by wa4557** [comment]
Cool, what next? Notification icons? Right now I have plenty of time :)

**Comment by dirtydancing** [comment]
Yeah, Ubuntu notification icons, what a treat :-) I have reported a mini series in https://github.com/numixproject/numix-icon-theme/issues/248, https://github.com/numixproject/numix-icon-theme/issues/251, https://github.com/numixproject/numix-icon-theme/issues/252, https://github.com/numixproject/numix-icon-theme/issues/253 and https://github.com/numixproject/numix-icon-theme/issues/255.

Currently, the Numix-Folders project is under active development. I have been asked to look into it. Not sure if this suits your taste, but you might check out https://github.com/numixproject/numix-folders/issues.

I will be offline for the next day or so, curious what you will come up with :-)

**Comment by dirtydancing** [comment]
After the latest PPA update including the merged https://github.com/numixproject/numix-icon-theme/pull/297, I believe this issue can be closed now.

**Comment by ChuckDaniels87** [comment]
@wa4557 Could you share your python script with us? It could be useful in the future for other issues. :stuck_out_tongue_winking_eye:

**Comment by wa4557** [comment]
@ChuckDaniels87 I'm afraid I don't have it any more, it was on my hdd (not on my ssd unfortunately) stopped working two days ago :( Frankly it was just a simple search and replace, in an already existing svg; nothing fancy. If I can recover data from my hdd i'll post it, otherwise I have to write something new the next time we need sth similar :-)

**Comment by ChuckDaniels87** [comment]
@wa4557 It's ok. I was curious about the script, we could collaborate to create a powerful tool to avoid tedious work. Sorry about your hdd lost. :dead:

**Comment by Foggalong** [comment]
Fixed in 290d12e2dd73c64e6aebfcbcff44c59f3350c78c
